gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -212,14 +215,17 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
         }
         catch (IOException e) {
             log.warn("Error extracting request body for rules processing: %s", e.getMessage());
+            parseErrorMessage = Optional.ofNullable(e.getMessage());
             isQueryParsingSuccessful = false;
         }
         catch (ParsingException e) {
             log.info("Could not parse request body as SQL: %s; Message: %s", body, e.getMessage());
+            parseErrorMessage = Optional.ofNullable(e.getMessage());
```

Comment:
Is `e.getMessage()` really nullable?

Reply:
Unless we have a logical bug, not really. 
Since parseErrorMessage is type `Optional<String>`, i couldn't directly save it as e.getMessage().
Any better ideas?

Reply:
Why not use `Optional.of()` method?